nativelib/src/main/resources/osx_platform.mm

```diff
@@ -0,0 +1,20 @@
+#ifdef __APPLE__
+#include <Foundation/Foundation.h>
+typedef const char *CString;
+
+extern "C" {
+void scalanative_mac_osx_version(int *major, int *minor, int *patch) {
```

Comment:
Lets make those to be `scalanative_os_version` and `scalanative_os_tmp_dir` with platform-specific dispatch logic inside. The platform-dependant code is better left on the C side (with ifdefs where appropriate), so that Scala side remains portable.